src/api/firebase.js

```diff
@@ -70,6 +70,52 @@ export function getItemData(snapshot) {
 	return filteredArray;
 }
 
+/**
+ * @param {array} itemArray the array of items to be sorted
+ * @returns {array} a reference to the array now sorted
+ * Will have all inactive items last, then sorted by days until next purchase, and items with same days sorted alphabetically
+ */
+export function comparePurchaseUrgency(itemArray) {
+	// split array to active and inactive, sort both, join back together and return
+	let today = new Date();
+
+	let activeArray = itemArray.filter(
+		(item) =>
+			item.dateLastPurchased === null ||
+			getDaysBetweenDates(today, item.dateLastPurchased.toDate()) < 60,
+	);
+	let inactiveArray = itemArray.filter(
+		(item) =>
+			item.dateLastPurchased !== null &&
+			getDaysBetweenDates(today, item.dateLastPurchased.toDate()) >= 60,
+	);
+
+	activeArray = sortByDaysUntilNextPurchase(activeArray);
+	inactiveArray = sortByDaysUntilNextPurchase(inactiveArray);
+
+	return activeArray.concat(inactiveArray);
+}
+
+/**
+ * @param {array} itemArray the array of items to be sorted
+ * @returns {array} a reference to the array now sorted
+ * Will be sorted with all by days until the item is expected to be purchased, with same number of days sorted alphabetically
+ */
+function sortByDaysUntilNextPurchase(itemArray) {
```

Comment:
Firebase also has an [`orderBy`](https://firebase.google.com/docs/firestore/query-data/order-limit-data) method which you can use to sort items so you wouldn't need to do a frontend calculation, but this also works!

Reply:
Wow I didn't even think about that, will definitely keep that in mind for future options, thanks! I think because I know that Firebase charges for queries I am subconsciously avoiding asking the database for information unless I absolutely have to. Definitely something I will watch out for in the future since optimizing for pricing could cause performance issues in other areas/situations.